pkgs/build-support/dart/pub2nix/package-graph/default.nix

```
@@ -0,0 +1,35 @@
+{
+  lib,
+  runCommand,
+  python3,
+}:
+
+{
+  name,
+  dependencies,
+  dependencySources,
+  pubspecFile,
+}:
+
+let
+  packages = lib.genAttrs dependencies (
+    dependency:
+    let
+      src = dependencySources.${dependency};
+    in
+    {
+      inherit src;
+      inherit (src) packageRoot;
+    }
+  );
+in
+(runCommand ((if name != null then "${name}-" else "") + "package-graph.json") {
+  inherit packages pubspecFile;
+
+  nativeBuildInputs = [ (python3.withPackages (ps: with ps; [ pyyaml ])) ];
```

```suggestion
  nativeBuildInputs = [ python3 python3Packages.pyyaml ];
```

Python packages get added to the PYTHONPATH through setup hooks, so there's no need to build a wrapper here.